modules/combobox/react/lib/Combobox.tsx

```diff
@@ -0,0 +1,348 @@
+    const lastItem = autoCompleteItemCount - 1;
+    let nextIndex = null;
+
+    switch (event.key) {
+      case 'ArrowUp':
+        const upIndex = currentIndex != null ? currentIndex - 1 : lastItem;
+        nextIndex = upIndex < 0 ? lastItem : upIndex;
+        break;
+
+      case 'ArrowDown':
+        const downIndex = currentIndex != null ? currentIndex + 1 : firstItem;
+        nextIndex = downIndex >= autoCompleteItemCount ? firstItem : downIndex;
+        break;
+
+      case 'Escape':
+        this.resetSearchInput();
+        break;
+
+      case 'Enter':
+        if (this.state.selectedAutocompleteIndex != null) {
+          const item = this.props.autocompleteItems[this.state.selectedAutocompleteIndex];
+          this.handleAutocompleteClick(event, item.props);
+          event.stopPropagation();
+          event.preventDefault();
+        }
+        break;
+
+      default:
+    }
+    this.setState({ selectedAutocompleteIndex: nextIndex });
+  };
+
+  handleSearchInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    if (this.props.onChange) {
+      this.props.onChange(event)
+    }
+    this.setState({ value: event.target.value });
+  };
+
+  isValidSingleChild = (child: React.ReactNode) => (React.isValidElement(child) && React.Children.only(child))
+
+  renderChildren = (child: React.ReactElement<TextInputProps>): React.ReactNode => {
+    if (!this.isValidSingleChild(child)) {
+      return child
+    }
+
+    return React.Children.map(child, (inputElement) => {
+      const newTextInputProps: Partial<TextInputProps> = {
+        id: this.props.id,
+        value: this.state.value,
+        inputRef: this.inputRef,
+        'aria-autocomplete': 'list',
+        'aria-controls': `${this.props.id}-${listBoxIdPart}`,
+        'aria-activedescendant': this.state.selectedAutocompleteIndex != null ? `${this.props.id}-${optionIdPart}-${this.state.selectedAutocompleteIndex}` : '',
+        onChange: this.handleSearchInputChange,
+        onKeyDown: this.handleKeyboardShortcuts,
+        onFocus: this.setFocus,
+        onBlur: this.setBlur,
+        css: this.props.showClearButton ? css({ paddingLeft: spacing.xl }) : {}
+      }
+      return React.cloneElement(inputElement, { ...inputElement.props, ...newTextInputProps });
+    })
+  }
+
+  render() {
+    const {
+      autocompleteItems,
+      children,
+      clearButtonType = IconButton.Types.Plain,
+      grow,
+      id,
+      initialValue,
+      onChange,
+      showClearButton,
+      ...elemProps
+    } = this.props;
+
+    return (
+      <Container
+        role='combobox'
+        aria-haspopup='listbox'
+        aria-owns={`${id}-${listBoxIdPart}`}
+        aria-expanded={autocompleteItems && autocompleteItems.length > 0}
+        {...elemProps}
+      >
+        <Status role="status" aria-live="polite">
+          {autocompleteItems ? this.state.showingAutocomplete && this.buildStatusString(autocompleteItems.length) : ''}
```

Comment:
This should only get read when showingAutocomplete toggles to true and autocompleteItems has changed so I think it will work correctly. Is there a specific case you are thinking of